### PR TITLE
Add explicit app arg for entity-crud-test and others

### PR DESCRIPTION
**Pull request by frenchy64**
Related https://github.com/threatgrid/iroh/issues/3986
Closes https://github.com/threatgrid/ctia/issues/930

Add an explicit app arg for `entity-crud-test` and add a `test-selected-stores-with-app` function that tests only a provided subset of store fixtures.

Also fixes a flaky logging test described by https://github.com/threatgrid/ctia/issues/930

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Add explicit app arg for entity-crud-test and others
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/test_helpers/store.clj [hunk 1]
```diff
@@ -11,13 +11,26 @@
...
         (fn []
           (t (helpers/get-current-app)))))))
+
+(s/defn test-for-each-store-with-app
+  "Takes a 1-argument function which accepts a Trapperkeeper `app`
+  which should succeed for all stores."
+  [t :- (s/=> s/Any
+              (s/named s/Any 'app))]
+  (test-selected-stores-with-app
+    (-> store-fixtures keys set)
```
Comment (vemv) [inline test/ctia/test_helpers/store.clj:35]: the two-space indentation, combined with the newline before the first arg makes it look as if `test-selected-stores-with-app` was a macro that took an arbitrary `& body`

Which makes `(-> store-fixtures keys set)` look like dead code :)

The following makes it more evident that we're looking at a vanilla function call:

```clj
(test-selected-stores-with-app (-> store-fixtures keys set)
                               t)
```

**Review by vemv — APPROVED** [review]
LGTM!

File: test/ctia/test_helpers/store.clj [hunk 2]
```diff
@@ -11,13 +11,26 @@
...
+(s/defn test-selected-stores-with-app
   "Takes a 1-argument function which accepts a Trapperkeeper `app`
-  which should succeed for all stores."
-  [t :- (s/=> s/Any
+  which should succeed for the stores named in the first argument."
+  [selected-stores
+   t :- (s/=> s/Any
               (s/named s/Any 'app))]
-  (doseq [[store-key fixtures] store-fixtures]
+  (assert (set? selected-stores))
```
Comment (ereteog) [inline test/ctia/test_helpers/store.clj:20]: I would rather add a schema check for selected-stores and assert here that's it's not empty. WDYT?

**Review by ereteog — APPROVED** [review]
LGTM

**Comment by frenchy64** [comment]
Thanks for the feedback!

**Pull request merged**